modules/extensions-support/src/main/java/org/mule/runtime/module/extension/api/loader/java/type/ExtensionParameter.java

```diff
@@ -79,23 +80,23 @@ default boolean shouldBeAdvertised() {
    * @return A {@code boolean} indicating whether the parameter is a required or not
    */
   default boolean isRequired() {
-    return !(isAnnotatedWith(Optional.class));
+    return !(isAnnotatedWith(Optional.class) || isAnnotatedWith(org.mule.sdk.api.annotation.param.Optional.class));
   }
 
   /**
    * @return The {@link java.util.Optional} default value of the operation
    */
   default java.util.Optional<String> defaultValue() {
-    java.util.Optional<String> optionalDefaultValue = java.util.Optional.empty();
     final java.util.Optional<Optional> annotation = getAnnotation(Optional.class);
     if (annotation.isPresent()) {
-      final Optional optionalAnnotation = annotation.get();
-      final String defaultValue = optionalAnnotation.defaultValue();
-      if (!defaultValue.equals(Optional.NULL)) {
-        optionalDefaultValue = java.util.Optional.of(defaultValue);
-      }
+      return getDefaultValue(annotation.get());
+    }
+    final java.util.Optional<org.mule.sdk.api.annotation.param.Optional> sdkAnnotation =
```

Comment:
If both are present, do you think we should use the value from the extensions-api annotation?

Reply:
You're right. The other way makes more sense to me. I'll change it